hibernate-core/src/main/java/org/hibernate/boot/internal/InFlightMetadataCollectorImpl.java

```diff
@@ -1854,8 +1854,8 @@ private void processSecondPasses(ArrayList<? extends SecondPass> secondPasses) {
 			return;
 		}
 
-		for ( SecondPass secondPass : secondPasses ) {
-			secondPass.doSecondPass( getEntityBindingMap() );
```

Comment:
Could you explain what's the point of this change?

Reply:
Sorry, I still didn't get it, the only difference I saw is for-each loop vs for loop, and for-each loop is more readable.

Reply:
A second pass can be added to the collection while iterating, which is why we need to use indexed iteration to avoid concurrent modification exceptions.

Reply:
After discussing this with @sebersole, I'll create another second pass queue instead of this change.